File: python/fibonacci.py
def memoFib(n):
    if n == 0:
        return []
    if n == 1:
        return [0]
    
    ans = [0, 1]
    for i in range(2, n):
        ans.append(ans[i-1] + ans[i-2])  # Use previously calculated values
    return ans

def main():
    n = int(input("Enter num of fibonacci numbers you want: "))
    result = memoFib(n)
    for num in result:
        print(num, end=' ')

if __name__ == "__main__":
    main()

File: haskell/fibonacci.hs
import GHC.Base (VecElem(Int16ElemRep))
fibonacci :: Int -> [Int]
fibonacci n
    | n == 0 = []
    | n == 1 = [0]
    | otherwise = [0, 1] ++ [fibonacci (n - 1) + fibonacci (n - 2)]
main :: IO ()
main = do
    print "Enter num of fibonacci numbers you want: "
    input <- getLine
    let num = read input :: Int
    print (fibonacci num)